src/main/java/com/redhat/red/build/koji/kerberos/KrbAuthenticator.java

```diff
@@ -152,26 +165,28 @@ public static KrbClient newClient( String krb5ConfFilename )
         return krbClient;
     }
 
-    public static TgtTicket getTgt( KrbClient krbClient, String keytab, String ccache, String principal, String password )
+    public static TgtTicket getTgt( KrbClient krbClient, String keytab, String ccache, String principal,
+                                    String password )
             throws KrbException
     {
-        TgtTicket tgt = null;
+        TgtTicket tgt;
 
         if ( keytab != null )
         {
-            tgt = krbClient.requestTgt( principal, keytab );
+            File keytabFile = new File( keytab );
+            tgt = krbClient.requestTgt( principal, keytabFile );
         }
         else if ( ccache != null )
         {
             try
             {
-                CredentialCache cc = krbClient.resolveCredCache( new File( ccache ) );
-                Credential cred = cc.getCredentials().get( 0 );
+                CredentialCache credCache = krbClient.resolveCredCache( new File( ccache ) );
+                Credential cred = findMatchingCredential( principal, credCache );
```

Comment:
This failed to look for the correct credential (matching the given user principal) in the ccache before.